h/traversal/contexts.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+"""
+"Context resource" objects for Pyramid views.
+
+Context objects are reusable components that represent "the context of a view"
+or "the subject of a view".
+
+They can do things like wrap a model object (or multiple model objects) and,
+since they have access to the ``request``  object, provide access to properties
+that you need the request to compute such as permissions, or route URLs for a
+wrapped model object. The view, or any code that the view calls and passes the
+context object into, can then make use of these properties.
+
+These context objects are returned by the root resources in
+:py:mod:`h.traversal.roots` if the route is configured to use one of those root
+factories (see the :py:mod:`h.traversal.roots` for usage).
+
+For such a route Pyramid will conveniently pass the found context object into
+the view callable as the ``context`` argument.
+
+"""
+from __future__ import unicode_literals
+
+from pyramid.security import DENY_ALL
+from pyramid.security import Allow
+from pyramid.security import principals_allowed_by_permission
+
+from h.models.organization import ORGANIZATION_DEFAULT_PUBID
```

Comment:
Right now, `h.traversal.contexts` and `h.traversal.resources` are of a manageable size, but I wonder if, in the future, we may want to decompose them into individual modules, one per *Root and *Context, potentially nested in subdirs — not now, that would be premature, but the specific `organization` import here caught my eye and made me think of it...